R/oauth-init.R

```diff
@@ -105,18 +105,26 @@ init_oauth2.0 <- function(endpoint, app, scope = NULL, user_params = NULL,
     } else {
       code <- oauth_listener(authorize_url, is_interactive)$code
     }
-  } else
+  }
   # Use authorisation code to get (temporary) access token
 
   # Send credentials using HTTP Basic or as parameters in the request body
   # See https://tools.ietf.org/html/rfc6749#section-2.3 (Client Authentication)
-  req_params <- list(
-    client_id = app$key,
-    redirect_uri = redirect_uri,
-    grant_type = "authorization_code",
-    code = code)
 
-  if(without_auth_req) req_params <- NULL
+
+  if(without_auth_req) {
+    req_params <- list(
+      client_id = app$key,
+      client_secret = app$secret,
+      # redirect_uri = redirect_uri,
+      grant_type = "client_credentials")
```

Comment:
As tested in another use case with Yelp API (see #384), we may not have to use this `if` solution to set `grant_type` as `grant_type = "client_credentials"` could be pass manually by users through `user_params = list(grant_type = "client_credentials")`. 
However, there is still `code = code` that we do not need a a *client credentials grant* type